zipline/pipeline/term.py

```diff
@@ -466,8 +401,24 @@ def _validate(self):
         if self.inputs is NotSpecified:
             raise TermInputsNotSpecified(termname=type(self).__name__)
 
-        if not self.outputs:
+        if self.outputs is NotSpecified:
+            pass
+        elif not self.outputs:
             raise TermOutputsEmpty(termname=type(self).__name__)
+        else:
+            # Raise an exception if there are any naming conflicts between the
+            # term's output names and certain attributes.
+            disallowed_names = [
+                attr for attr in dir(ComputableTerm)
+                if not attr.startswith('_')
+            ]
+            for output in self.outputs:
+                if output.startswith('_') or output in disallowed_names:
+                    raise InvalidOutputName(
+                        output_name=output,
+                        termname=type(self).__name__,
+                        disallowed_names=disallowed_names,
```

Comment:
Note that `compute` is not in this list, so will still error if used as an output name. Without trying to make this too muddy, I'm thinking it should be added explicitly to the disallowed list